packages/records/lib/models/records.ts

```diff
@@ -545,6 +577,17 @@ export async function update({
                     const updated = await query;
                     updatedKeys.push(...updated.map((record) => record.external_id));
 
+                    const firstUpdateThisMonth = updated.filter((r) => {
+                        const firstDayOfMonth = dayjs().startOf('month').toDate();
+                        const oldRecord = oldRecords.find((old) => old.external_id === r.external_id);
+                        if (!oldRecord) {
+                            return false;
+                        }
+                        const previousLastModifiedAt = dayjs(oldRecord.updated_at);
+                        return previousLastModifiedAt.isBefore(firstDayOfMonth);
+                    });
+                    billedKeys.push(...firstUpdateThisMonth.map((r) => r.external_id));
```

Comment:
why not do a regular for loop that pushes directly to billedKeys? you are iterating twice over the array rn